supernode/daemon/mgr/cdn/super_writer_util.go

```diff
@@ -0,0 +1,99 @@
+package cdn
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"sync"
+
+	"github.com/dragonflyoss/Dragonfly/supernode/config"
+	"github.com/dragonflyoss/Dragonfly/supernode/store"
+
+	"github.com/sirupsen/logrus"
+)
+
+func calculateRoutineCount(httpFileLength int64, pieceSize int32) int {
+	routineSize := config.CDNWriterRoutineLimit
+	if httpFileLength <= 0 || pieceSize <= 0 {
```

Comment:
When `httpFileLength` equals 0, it means that this is a empty file, it just need 1 goroutine to write into storage.